File: beginner/1018/Banknotes.java
// challenge link - https://judge.beecrowd.com/en/problems/view/1018

import java.io.IOException;
import java.util.Scanner;

public class Banknotes {
	public static void main(String[] args) throws IOException {
		Scanner reader = new Scanner(System.in);
		int value = reader.nextInt();
		int ogValue = value;
		
		int[] notes = {100, 50, 20, 10, 5, 2, 1};
		int[] notesUsed = new int[notes.length];
		int notes_index = 0;

		while (notes_index < notes.length) {
			if (value >= notes[notes_index]) {
				value -= notes[notes_index];
				notesUsed[notes_index] = notesUsed[notes_index] + 1;
			} else {
				notes_index++;
			}
		}
		
		System.out.println(ogValue);

		for (int i = 0; i < notes.length; i++) {
			System.out.println(notesUsed[i] + " nota(s) de R$ " + notes[i] + ",00");
		}
	}
}

File: beginner/1018/banknotes.c
// challenge link - https://judge.beecrowd.com/en/problems/view/1018

#include <stdio.h>

int main(void) {
	int value = 0;
	scanf("%d", &value);

	int bank_notes[] = {100, 50, 20, 10, 5, 2, 1};
	int amount_of_notes = 7;
	int used = 0;
	int i = 0;

	printf("%d\n", value);

	while (i < amount_of_notes) {
		if (bank_notes[i] <= value) {
			used++;
			value -= bank_notes[i];
		} else {
			printf("%d notas(s) de R$ %d,00\n", used, bank_notes[i]);
			used = 0;
			i++;
		}
	}
}
